plugin/db/mysql/dump.go

```diff
@@ -226,26 +234,67 @@ func dumpTxn(ctx context.Context, txn *sql.Tx, database string, out io.Writer, s
 			}
 		}
 
+		// Disable foreign key check.
+		// mysqldump uses the same mechanism. When there is any schema or data dependency, we have to disable
+		// the unique and foreign key check so that the restoring will not fail.
+		if _, err := io.WriteString(out, disableUniqueAndForeignKeyCheckStmt); err != nil {
+			return err
+		}
+
 		// Table and view statement.
+		// We have to dump the table before views because of the structure dependency.
 		tables, err := getTablesTx(txn, dbName)
 		if err != nil {
 			return errors.Wrapf(err, "failed to get tables of database %q", dbName)
 		}
+		// Construct temporal views.
+		// Create a table with the same name as the view and with columns of
+		// the same name in order to satisfy views that depend on this view.
+		// The table will be removed when the actual view is created.
+		// The properties of each column, are not preserved in this temporary
+		// table, because they are not necessary.
+		// This will not be necessary once we can determine dependencies
+		// between views and can simply dump them in the appropriate order.
```

Comment:
```suggestion
		// Create a temporary view with the same name as the view and with columns of
		// the same name in order to satisfy views that depend on this view.
		// This temporary view will be removed when the actual view is created.
		// The properties of each column, are not preserved in this temporary
		// view. They are not necessary because other views only need to reference
		// the column name, thus we generate SELECT 1 AS colName1, 1 AS colName2.
		// This will not be necessary once we can determine dependencies
		// between views and can simply dump them in the appropriate order.
```

Reply:
Attempted a rewrite. PTAL